### ISSUE TITLE
Raspi connection schematic

### ISSUE THREAD

**Issue by tomburtonwood** [ArduCAM/RaspberryPi#7]
Hi,

Sorry for cross posting. i just realized i posted in the Arduino section. Apologies. So as i wrote previously i have bought the four camera module and would like to connect it to a Raspberry Pi as shown on this page ( http://www.arducam.com/arducam-porting-raspberry-pi/arducam-pizero-3/ ), but i'm a total noob when it comes to all things Pi. 

Can you please provide a schematic / drawing / tutorial showing how to connect the Pi to the module and how run any associated code. Any help would be much appreciate as i'm hoping to get my project going asap - and if this works - i can then order a bunch more.

Thank you!

T

**Comment by ArduCAM** [comment]
@Tom, Please read the guide below: http://www.arducam.com/downloads/shields/simple_guide_V2.pdf  , let me know if you need more help.

**Comment by tomburtonwood** [comment]
I have a pi2 will this pi zero wiring diagram and code work the same?

 typos are mine

**Comment by tomburtonwood** [comment]
Also does the 4 connector board wire up the same as the single camera?

 typos are mine

**Comment by ArduCAM** [comment]
@Tom, all Pi model B has the same pinout, so they are the same. You can connect 1 ~ 4 cameras as our example shows.

**Comment by dmolner** [comment]
Hello, 

Is possible use this for connect more one càmera? 
How select every cam? 

Best Regards, 
David

**Comment by ArduCAM** [comment]
@dmolner Please read the guide below: http://www.arducam.com/downloads/shields/simple_guide_V2.pdf , let me know if you need more help.

**Comment by dmolner** [comment]
Great,
Is possible connect different cameras?
I have two Arducam one 2Mp and other 5Mp.

**Comment by ArduCAM** [comment]
@dmolner, Yes, it is possible. But you have to initialize all cameras seperately, and after that everything is the same for all cameras.

**Comment by sinabagci** [comment]
Hello,

![image](https://cloud.githubusercontent.com/assets/20249301/18005944/c0bb78a8-6ba5-11e6-93e0-17688ce8df9d.png)

When I tried to open jpg file, it gives error like "unable to open file"

Didn't manage to find out whats wrong

**Comment by sinabagci** [comment]
Finally I find out the problem. All thanks to terrible documentation of ArduCAM..

Simple_GuideV2's CS pin wiring is wrong. Correct wiring is in WiringPiI2C.h

The Correct Pin for CS is "BCM:17" "wiringPi:0"

![image](https://cloud.githubusercontent.com/assets/20249301/18013256/f3d41c34-6bc5-11e6-937d-b46d2241a0ba.png)


![image](https://cloud.githubusercontent.com/assets/20249301/18013234/e06c78d0-6bc5-11e6-811b-cef498dbdf4f.png)

**Comment by zippy123** [comment]
> The Correct Pin for CS is "BCM:17" "wiringPi:0"
Yeah, that fixed it for me too.  Thanks!

**Comment by mikewallis** [comment]
This isn't working for me. If I follow the wiring diagram I get 

```
pi@raspberrypi ~ $ ov5642_capture -c test.jpg 640x480
SPI interface OK!
OV5642 detected
Changed resolution1 to 640x480
Start capture
 CAM1 Capture Done
Reading FIFO
pi@raspberrypi ~ $ file test.jpg 
test.jpg: data
```

...and the file is garbage. If I follow sinabagci's advice above (swapping the connection from pin 24 to pin 17) I get a SPI Interface error and the file never gets written. At the moment the cam is unusable. 

Any help with this would be appreciated.

**Comment by zippy123** [comment]
What values are in the file?  

Sent from iPhone, explaining the odd terse syntax. 

Dave

**Comment by mikewallis** [comment]
A whole bunch of 'U's, in a single line. No control chars, no hex, no EXIF header, no spaces.

**Comment by zippy123** [comment]
Here's my wiring (not sure if you can follow) but I took it apart just yesterday.  It was working.  Hope this might help but I can't recall any other details that would help.  

![img_0303](https://cloud.githubusercontent.com/assets/5561248/19017102/9cd281e8-87e3-11e6-807a-82437732e783.JPG)
![img_0304](https://cloud.githubusercontent.com/assets/5561248/19017103/a37ae0e4-87e3-11e6-8d50-f66e73042dd1.JPG)

**Comment by zippy123** [comment]
I also found [this site](https://pinout.xyz) helpful and accurate.

**Comment by mikewallis** [comment]
That's really helpful! Thanks, Dave, I really appreciate that. (I'm getting a different JPEG error but at least it's writing the right headers now...)

**Comment by ArduCAM** [comment]
@mikewalls, please attach the captured files, we will try to help you figure out what's the problem.

**Comment by pidai** [comment]
This isn't working for me..
I tried different wirings, but i get everytime the same result:
![2017-01-26-091514_1824x984_scrot](https://cloud.githubusercontent.com/assets/25107860/22326044/72cbd8f2-e3b1-11e6-9f61-799f51a42d16.png)
It starts to capture, but then it does nothing. I've waited more than half an hour but it didn't take a pic.

Phil

**Comment by deniseymoran** [comment]
Here is a wiring diagram that might help:
![498_wiring - page 1](https://cloud.githubusercontent.com/assets/11726781/23336287/f3f7345e-fb7f-11e6-8014-ed75ecd79647.png)

I was having the same problem where test.jpg was in the incorrect format (just rows of 'U's), and this forum solved the problem because the wiring was wrong in the simple guide. I'm using an arducam OV2640 with Raspberry Pi 3 B. Thanks for the solution and I hope everyone can get theirs to work!

**Comment by deniseymoran** [comment]
Here's a correct wiring diagram to connect the arducam to raspberry pi 3 B that might help:

![498_wiring - page 1](https://cloud.githubusercontent.com/assets/11726781/23336295/431cff0a-fb80-11e6-8163-406bce1d7a08.png)

**Comment by KoRba88** [comment]
pidai in memorysaver.h comment ov2640 and uncomment ov5642

**Comment by 813584** [comment]
To sell this camera with a ribbon cable and state it works with the raspberry pi and include a set of instructions that state activating the interface via raspi-config and a reboot a voila! it works is nothing less than a lie. Even the tutorial for Raspberry Pi on the Arducam website is only a template for the web page.
